components/omega/src/infra/OmegaKokkos.h

```diff
@@ -0,0 +1,122 @@
+#ifndef OMEGA_KOKKOS_H
+#define OMEGA_KOKKOS_H
+//===-- base/OmegaKokkos.h - Omega extension of Kokkos ------*- C++ -*-===//
+//
+/// \file
+/// \brief Extends Kokkos for Omega
+///
+/// This header extends Kokkos for Omega.
+//
+//===----------------------------------------------------------------------===//
+
+#include "DataTypes.h"
+
+namespace OMEGA {
+
+#define OMEGA_SCOPE(a, b) auto &a = b
+
+using ExecSpace     = MemSpace::execution_space;
+using HostExecSpace = HostMemSpace::execution_space;
+
+#ifdef OMEGA_TARGET_DEVICE
+
+template <typename V>
+auto createHostCopy(const V &view)
+    -> Kokkos::View<typename V::data_type, HostMemLayout, HostMemSpace> {
+   return Kokkos::create_mirror_view_and_copy(HostExecSpace(), view);
+}
+
+template <typename V>
+auto createDeviceCopy(const V &view)
+    -> Kokkos::View<typename V::data_type, MemLayout, MemSpace> {
+   return Kokkos::create_mirror_view_and_copy(ExecSpace(), view);
+}
+
+#else
+
+template <typename V> V createHostCopy(const V &view) { return view; }
+
+template <typename V> V createDeviceCopy(const V &view) { return view; }
+
+#endif
+
+// function alias to follow Camel Naming Convention
+template <typename D, typename S> void deepCopy(D &dst, const S &src) {
+   Kokkos::deep_copy(dst, src);
+}
+
+template <typename E, typename D, typename S>
+void deepCopy(E &space, D &dst, const S &src) {
+   Kokkos::deep_copy(space, dst, src);
+}
+
+#if OMEGA_LAYOUT_RIGHT
+
+template <int N, class... Args>
+using Bounds = Kokkos::MDRangePolicy<
+    ExecSpace, Kokkos::Rank<N, Kokkos::Iterate::Right, Kokkos::Iterate::Right>,
+    Args...>;
+
+#elif OMEGA_LAYOUT_LEFT
+
+template <int N, class... Args>
+using Bounds = Kokkos::MDRangePolicy<
+    ExecSpace, Kokkos::Rank<N, Kokkos::Iterate::Left, Kokkos::Iterate::Left>,
+    Args...>;
+
+#else
+
+#error "OMEGA Memory Layout is not defined."
+
+#endif
+
+// parallelFor: with label
+template <int N, class F, class... Args>
```

Comment:
@mwarusz, My intention here is to give users the opportunity to specify optional type arguments for MDRangePolicy or RangePolicy. For now, I am not sure if this is useful or needed at all. However, I thought it could be one of the future-proof approaches. What do you think?

Reply:
I agree that this is desirable. I am just wondering if we can create a better interface. For example, with the current version if you want to use a lambda and specify `Kokkos::IndexType<long>` in a `parallelFor` you need to do:
```cpp
auto F = KOKKOS_LAMBDA (long I) { a(I) = I; };
parallelFor<1, decltype(F), Kokkos::IndexType<long>>("label", {N}, F);
```
That is, you need to create the lambda first and then get its type using `decltype`. Maybe for advanced uses like this we would always create named functors ?

Reply:
I might understand your concern now. To specify "Args", we must first define the named functor before "Args". To achieve this, we need to define the functor before "parallelFor<...>()", which necessitates complex and strict usage. If this is the case, for now, I am unsure how to design a better interface. Because it may require time for developers to begin utilizing this advanced feature, we may consider taking our time to contemplate a new interface.